pkg/types/gcp/validation/platform_test.go

```diff
@@ -153,6 +153,88 @@ func TestValidatePlatform(t *testing.T) {
 			credentialsMode: types.MintCredentialsMode,
 			valid:           false,
 		},
+		{
+			name: "invalid gcp endpoint blank name",
+			platform: &gcp.Platform{
+				Region: "us-east1",
+				ServiceEndpoints: []gcp.ServiceEndpoint{
+					{
+						Name: "",
+						URL:  "https://my-custom-endpoint.example.com/copmute/v1/",
+					},
+				},
+			},
+			valid: false,
+		},
+		{
+			name: "invalid gcp endpoint invalid name",
+			platform: &gcp.Platform{
+				Region: "us-east1",
+				ServiceEndpoints: []gcp.ServiceEndpoint{
+					{
+						Name: "badname",
+						URL:  "https://my-custom-endpoint.example.com/copmute/v1/",
+					},
+				},
+			},
+			valid: false,
+		},
+		{
+			name: "invalid gcp endpoint duplicate name",
+			platform: &gcp.Platform{
+				Region: "us-east1",
+				ServiceEndpoints: []gcp.ServiceEndpoint{
+					{
+						Name: "compute",
+						URL:  "https://my-custom-endpoint.example.com/compute/v1/",
+					},
+					{
+						Name: "compute",
+						URL:  "https://my-custom-endpoint.example.com/compute/v2/",
+					},
+				},
+			},
+			valid: false,
+		},
+		{
+			name: "invalid gcp endpoint url blank",
+			platform: &gcp.Platform{
+				Region: "us-east1",
+				ServiceEndpoints: []gcp.ServiceEndpoint{
+					{
+						Name: "compute",
+						URL:  "",
+					},
+				},
+			},
+			valid: false,
+		},
+		{
+			name: "invalid scheme gcp endpoint url",
+			platform: &gcp.Platform{
+				Region: "us-east1",
+				ServiceEndpoints: []gcp.ServiceEndpoint{
+					{
+						Name: "compute",
+						URL:  "http://my-custom-endpoint.example.com/compute/v1/",
+					},
+				},
+			},
+			valid: false,
+		},
```

Comment:
nit: should we have a test for the following cases:
-  `/compute/v1`: relative url will return error with missing hostname
- `my-custom-endpoint.example.com/compute/v1`: url without scheme will be appended `https` -> valid.